pkg/policy/api/rule.go

```diff
@@ -0,0 +1,234 @@
+
+	// Comment is a free form string, it can be used by the creator of the
+	// rule to store human readable explanation of the purpose of this
+	// rule. Rules cannot be identified by comment.
+	//
+	// +optional
+	Comment string `json:"comment,omitempty"`
+}
+
+// IngressRule contains all rule types which can be applied at ingress,
+// i.e. network traffic that originates outside of the endpoint and
+// is entering the endpoint selected by the endpointSelector.
+//
+// - All members of this structure are optional.
+// - All rule types are considered independently
+type IngressRule struct {
+	// FromEndpoints is a list of endpoints identified by an
+	// EndpointSelector which are allowed to communicate with the endpoint
+	// subject to the rule.
+	//
+	// If omitted and no other IngressRule applies a FromEndpoints rule on
+	// the endpoint, the endpoint is only allowed to take connections from
+	// endpoints which are whitelisted in fromCIDR.
+	//
+	// +optional
+	FromEndpoints []EndpointSelector `json:"fromEndpoints,omitempty"`
+
+	// FromRequires is a selector which must match in combination with at
+	// least one selector defined in FromEndpoints. The selector must match
+	// in any allowed source but by itself does not grant access.  This
+	// allows to define a rule which says, allow "foo" to "bar" but only if
+	// "group=A" is set while "group=A" as a label itself has no immediate
+	// access rights.
+	//
+	// +optional
+	FromRequires []EndpointSelector `json:"fromRequires,omitempty"`
+
+	// ToPorts is a list of destination ports identified by port number and
+	// protocol which the endpoint subject to the rule is allowed to
+	// receive connections on.
+	//
+	// If omitted and no other IngressRule applies a ToPorts rule on the
+	// endpoint, the endpoint is allowed to receive connections on all
+	// ports.
+	//
+	// +optional
+	ToPorts []PortRule `json:"toPorts,omitempty"`
+
+	// FromCIDR is a list of IP blocks which the endpoint subject to the
+	// rule is allowed to receive connections from. This will match on the
+	// source IP address of incoming connections.
+	//
+	// This restriction does not apply to connections from endpoints which
+	// have been whitelisted using a FromEndpoints rule of an IngressRule.
+	//
+	// If omitted and no other IngressRule applies a fromCIDR rule on the
+	// endpoint, the endpoint is only allowed to take connections from
+	// endpoints which are whitelisted in FromEndpoints.
+	//
+	// +optional
+	FromCIDR []CIDR `json:"toCIDR,omitempty"`
+}
+
+// EgressRule contains all rule types which can be applied at egress, i.e.
+// network traffic that originates inside the endpoint and exits the endpoint
+// selected by the endpointSelector.
+//
+type EgressRule struct {
+	// ToPorts is a list of destination ports identified by port number and
+	// protocol which the endpoint subject to the rule is allowed to
+	// connect to.
+	//
+	// If omitted and no other IngressRule applies a ToPorts rule on the
```

Comment:
I meant EgressRule but I believe the paragraph is not required at all for understanding.